### PR TITLE
plugin/kubernetes: Add support for dual stack ClusterIP Services

### PR DESCRIPTION
**Pull request by chrisohaver**
Signed-off-by: Chris O'Haver <cohaver@infoblox.com>

### 1. Why is this pull request needed and what does it do?

Kubernetes 1.20 added support for dual stack Cluster IP services, such that a Cluster IP Service can now have two IP addresses (1 v4 and 1 v6).

### 2. Which issues (if any) are related?

replaces #4246

### CODE REVIEW

File: plugin/kubernetes/object/service.go [hunk 1]
```diff
@@ -40,13 +40,19 @@ func ToService(obj meta.Object) (meta.Object, error) {
...
 		Index:        ServiceKey(svc.GetName(), svc.GetNamespace()),
-		ClusterIP:    svc.Spec.ClusterIP,
 		Type:         svc.Spec.Type,
 		ExternalName: svc.Spec.ExternalName,
 
 		ExternalIPs: make([]string, len(svc.Status.LoadBalancer.Ingress)+len(svc.Spec.ExternalIPs)),
 	}
 
+	if len(svc.Spec.ClusterIPs) > 0 {
+		s.ClusterIPs = make([]string, len(svc.Spec.Ports))
```
Comment (aojea) [inline plugin/kubernetes/object/service.go]: why  `len(svc.Spec.Ports))` instead of ` len(svc.Spec.ClusterIPs))` ??

Comment (chrisohaver) [reply plugin/kubernetes/object/service.go]: oops. Thats a mistake. Thanks!

File: plugin/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -502,10 +501,11 @@ func (k *Kubernetes) findServices(r recordRequest, zone string) (services []msg.
 
 			err = nil
 
-			s := msg.Service{Host: svc.ClusterIP, Port: int(p.Port), TTL: k.ttl}
-			s.Key = strings.Join([]string{zonePath, Svc, svc.Namespace, svc.Name}, "/")
-
-			services = append(services, s)
+			for _, ip := range svc.ClusterIPs {
```
Comment (zouyee) [inline plugin/kubernetes/kubernetes.go:504]: If the dual stack feature is not enabled, do we need to preserve signal ClusterIP processing？

Comment (chrisohaver) [reply plugin/kubernetes/kubernetes.go:504]: If dual stack is not enabled, ClusterIPs will contain the single ClusterIP as the first element.

**Comment by chrisohaver** [comment]
I'm having trouble creating a functioning dual-stack k8s cluster to actually test these changes.

**Comment by aojea** [comment]
> I'm having trouble creating a functioning dual-stack k8s cluster to actually test these changes.

@chrisohaver https://github.com/aojea/kind/releases/tag/dualstack
ping me in slack if it still doesn´t work

**Comment by chrisohaver** [comment]
@aojea, Thanks so much! That was _exactly_ what I needed.

I created a dual-stack clusterIP service, and the coredns responds as expected when testing A, AAAA, and SRV.

**Comment by aojea** [comment]
great LGTM then :smile:

**Review by yongtang — APPROVED** [review]
lgtm

**Pull request merged**